File: trunk/ShadowFrameworkLite20/src/shadow/pipeline/SFPrimitiveIndices.java
package shadow.pipeline;


/**
 * The Description of All the Indices Being part of a Primitive
 * 
 * @author Alessandro Martinelli
 */
public class SFPrimitiveIndices {

	private int primitiveIndices[];
	
	public SFPrimitiveIndices(){
		
	}
	
	public SFPrimitiveIndices(SFPrimitive primitive){
		this.primitiveIndices = new int[primitive.getIndicesCount()];	
	}
	
	public void set(SFPrimitiveIndices indices){
		for (int i = 0; i < primitiveIndices.length; i++) {
			primitiveIndices[i]=indices.primitiveIndices[i];	
		}
	}

	public int[] getPrimitiveIndices() {
		return primitiveIndices;
	}

	public void setPrimitiveIndices(int[] primitiveIndices) {
		this.primitiveIndices = primitiveIndices;
	}

	public void setData(SFPrimitiveIndices indices,int firstIndex,int size){
		for (int j = firstIndex; j < firstIndex+size; j++) {
			primitiveIndices[j]=indices.primitiveIndices[j];
		}
	}
	
	public SFPrimitiveIndices clone(){
		SFPrimitiveIndices indices=new SFPrimitiveIndices();
		indices.primitiveIndices=new int[primitiveIndices.length];
		return indices;
	}
}

File: SFDataApplicationTestC++/shadow/pipeline/SFPrimitiveIndices.cpp

//
//  SFPrimitiveIndices.cpp
//  
//
//  Created by Alessandro Martinelli on 18/10/12.
//  Copyright (c) 2012 Alessandro Martinelli. All rights reserved.
//

#include "SFPrimitiveIndices.h"


namespace sf{

SFPrimitiveIndices::SFPrimitiveIndices(){
    this->size=0;
    this->primitiveIndices=0;
}

SFPrimitiveIndices::SFPrimitiveIndices(SFPrimitive* primitive){
    //cout << "primitive->getIndicesCount() " << primitive->getIndicesCount() << endl ;
    this->primitiveIndices = new int[primitive->getIndicesCount()];
    this->size=primitive->getIndicesCount();
}

void SFPrimitiveIndices::set(SFPrimitiveIndices* indices){
    for (int i = 0; i < size; i++) {
        primitiveIndices[i]=indices->primitiveIndices[i];
    }
}

int SFPrimitiveIndices::length(){
    return size;
}

int* SFPrimitiveIndices::getPrimitiveIndices() {
    return primitiveIndices;
}

void SFPrimitiveIndices::setPrimitiveIndices(int* primitiveIndices) {
    this->primitiveIndices = primitiveIndices;
}

void SFPrimitiveIndices::setData(SFPrimitiveIndices indices,int firstIndex,int size){
    for (int j = firstIndex; j < firstIndex+size; j++) {
        primitiveIndices[j]=indices.primitiveIndices[j];
    }
}

/*SFPrimitiveIndices* SFPrimitiveIndices::clone(){
    SFPrimitiveIndices* indices=new SFPrimitiveIndices();
    indices->primitiveIndices=new int[size];
    return indices;
}*/
}
